File: src/lib/combos.ts
import { filterPresentRows, parseVariable } from "./helpers.ts"
import { Block, Combo } from "./types.ts"

export function parseCombos(block: Block): Combo[] {
  const { lines, indexToLineNr } = filterPresentRows(block)

  const combos: Combo[] = []

  lines.forEach((line, index) => {
    const v = parseVariable(line)
    if (v != null) {
      combos.push({ keys: v.name, action: v.value, lineNr: indexToLineNr[index] })
    }
  })

  return combos
}

File: internal/parser/combos.go
package parser

import "github.com/elentok/qmkmd/internal/layout"

func ParseCombos(block layout.Block) []layout.Combo {
	present := layout.FilterPresentRows(block)
	combos := make([]layout.Combo, 0, len(present.Lines))

	for i, line := range present.Lines {
		name, value, ok := layout.ParseVariable(line)
		if !ok {
			continue
		}
		combos = append(combos, layout.Combo{
			Keys:   name,
			Action: value,
			LineNr: present.IndexToLineNr[i],
		})
	}

	return combos
}
